### PR TITLE
Simplified JpaRolloutManagement, moved rollout handling logic to JpaRolloutHandler

### PR DESCRIPTION
**Pull request by a-sayyed**
The current `JpaRolloutManagement` also acts as a `RolloutHandler`, which is confusing from the 'single point of responsibility' point of view.
This PR moves the rollout-handling logic outside of `JpaRolloutManagement`, and into `JpaRolloutHandler`

### CODE REVIEW

**Review by StefanKlt — APPROVED** [review]
Looks good thanks @a-sayyed

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/RolloutHandler.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import static org.eclipse.hawkbit.im.authentication.SpPermission.SpringEvalExpressions;
+
+import org.eclipse.hawkbit.repository.model.Target;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * Represents the handler service for creating, deleting, and starting a Rollout
+ */
+@FunctionalInterface
+public interface RolloutHandler {
```
Comment (a-sayyed) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/RolloutHandler.java:20]: allign with `AutoAssignmentExecutor`, rename

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutExecutor.java [hunk 2]
```diff
@@ -0,0 +1,660 @@
+/**
```
Comment (stefbehl) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutExecutor.java:1]: Did you change the logic in any way or has the code been moved as is?

Comment (a-sayyed) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutExecutor.java:1]: no, everything was just moved as is, reference to the `start` method on the `RolloutManagement` is called via `context.getBean(RolloutManagement.class).start()` see https://github.com/eclipse/hawkbit/blob/4029dbfaa36908ea5a89410e252e89b9bff29f3a/hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutExecutor.java#L255

**Review by stefbehl — APPROVED** [review]
Looks good. Thanks Ahmed!

**Pull request merged**